internal/engine/event_watch_manager.go

```diff
@@ -33,18 +34,21 @@ type EventWatchManager struct {
 	uidMap   map[k8s.UID]uidMapEntry
 	uidMapMu sync.RWMutex
 	clock    clockwork.Clock
+	f        feature.Feature
 }
 
-func NewEventWatchManager(kClient k8s.Client, clock clockwork.Clock) *EventWatchManager {
+func NewEventWatchManager(kClient k8s.Client, clock clockwork.Clock, f feature.Feature) *EventWatchManager {
 	return &EventWatchManager{
 		kClient: kClient,
 		uidMap:  make(map[k8s.UID]uidMapEntry),
 		clock:   clock,
+		f:       f,
 	}
 }
 
 func (m *EventWatchManager) needsWatch(st store.RStore) bool {
-	if !k8sEventsFeatureFlagOn() {
+	enabled, err := m.f.IsEnabled("events")
+	if !enabled || err != nil {
 		return false
```

Comment:
We should probably also handle the case of enabled going from true to false (i.e., stop the watch).